src/constraints/DateConstraints.ts

```diff
@@ -3,7 +3,10 @@ import { Result } from '../lib/Result';
 import type { IConstraint } from './base/IConstraint';
 import { Comparator, eq, ge, gt, le, lt, ne } from './util/operators';
 
-function dateComparator(comparator: Comparator, name: string, expected: string, number: number): IConstraint<Date> {
+export type DateComparatorConstraintName = `s.date.${'lt' | 'le' | 'gt' | 'ge' | 'eq' | 'ne'}`;
+export type DateConstraintName = DateComparatorConstraintName | `s.date.${'eq(NaN)' | 'ne(NaN)'}`;
```

Comment:
Kyra are you okay? Same goes for all the other ones where you have 2 types.

![19-49-25_03-02-2022_vivaldi](https://user-images.githubusercontent.com/4019718/152409133-efb8e48a-0946-4148-948e-0c4595260e05.png)

![19-49-30_03-02-2022_vivaldi](https://user-images.githubusercontent.com/4019718/152409139-e095abc0-3ebd-4a92-aa0d-0371a27e7a2f.png)

![19-50-12_03-02-2022_ApplicationFrameHost](https://user-images.githubusercontent.com/4019718/152409149-cbd1de60-9da0-4927-936b-df8c866ce5fe.png)